### PR TITLE
fix/6326

### LINKED ISSUE #6326: web admin(RADIUS filters): condition on connection_sub_type doesn't work as expected
**Describe the bug**
When I create a RADIUS filter like this:

Capture d’écran de 2021-04-26 09-02-30

I got following config in `radius_filters.conf`:

```
condition=connection_type == "Wireless-802.11-EAP" && connection_sub_type == "EAP-TLS"
```

This condition doesn't work.

**To Reproduce**
1. Create RADIUS filter seen in screenshot
2. Forge a EAP-TLS RADIUS request (on Wireless)

=> No match.

**Expected behavior**
Filter should match.

**Additional context**
- If I replace "EAP-TLS" in `radius_filters.conf` by "13", my filter works.
- connection_sub_type support in RADIUS filter has been added recently by https://github.com/inverse-inc/packetfence/commit/821dd75ce3fa6b484bd6d94b8c5aa780f7f017fd .

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description

Normalize connection_sub_type to use the numeric value.

# Impacts
Radius filters

# Issue
fixes #6326

# Delete branch after merge
NO

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Bug Fixes
* Normalize connection_sub_type to use the numeric value (#6326)

### CODE REVIEW

File: lib/pf/factory/condition.pm [hunk 1]
```diff
@@ -158,10 +162,35 @@ sub _build_parent_condition {
...
+    my $last_key = $keys[-1];
+    if (exists $VALUE_FILTERS{$last_key}) {
+        $val = $VALUE_FILTERS{$last_key}->($val);
+    }
+
+    my $sub_condition = $BINARY_OP{$op}->new(value => $val);
+    return build_parent_condition($sub_condition, @keys);
+}
+
+sub normalize_eap_type {
```
Comment (nqb) [inline lib/pf/factory/condition.pm:181]: This function doesn't seem to be used.

File: lib/pf/config/builder/filter_engine.pm [hunk 2]
```diff
@@ -118,20 +118,31 @@ build a filter
...
+    }
 }
 
+=head2 rewriteConditions
+
+Rewrite Conditions
+
+=cut
+
+sub rewriteConditions {
```
Comment (nqb) [inline lib/pf/config/builder/filter_engine.pm]: Is this function really necessary ?

In my understanding, this function returns what it received.

**Review by nqb — CHANGES_REQUESTED** [review]
Few remarks but fix work as expected.

I got following content in my `radius_filters.conf`:

```
[test_bugfix]
description=test_bugfix
status=enabled
condition=connection_type == "Wireless-802.11-EAP" && connection_sub_type == "EAP-TLS"
merge_answer=no
answer.0=request:Reply-Message = test_bugfix
```

and I can see that my RADIUS request is matched by this filter without changing `connection_sub_type`.

**Comment by nqb** [comment]
No more changes to come @jrouzierinverse (see my second comment) ?

File: lib/pf/factory/condition.pm (hunk 1 above)
Comment (jrouzierinverse) [reply lib/pf/factory/condition.pm:181]: This is for a future feature #6485

**Pull request merged**